Makefile

```diff
@@ -146,11 +146,18 @@ endif
 
 # setup opencv
 ifeq ($(USE_OPENCV), 1)
-	CFLAGS += -DMXNET_USE_OPENCV=1 $(shell pkg-config --cflags opencv)
-	LDFLAGS += $(filter-out -lopencv_ts, $(shell pkg-config --libs opencv))
+	ifeq ("$(shell pkg-config --exists opencv4; echo $$?)", "0")
```

Comment:
there needs to be a way for user of make to override whatever is found by pkg-config.

Reply:
not a requirement but nice to have. libturbojpeg has two variables for this purpose: https://github.com/apache/incubator-mxnet/blob/master/Makefile#L460-L461
Also, many of what's returned in pkg-config of opencv are not necessary for mxnet. The only necessary modules in opencv for mxnet are core, imgcodecs, and imgproc. As a result, I turn off most of the build options for opencv in mxnet pip package: https://github.com/apache/incubator-mxnet/blob/master/tools/dependencies/opencv.sh#L39

Reply:
I see. I will modify it. Thanks!

Reply:
OpenCV2 doesn't have libopencv_imgcodecs.so, and I do not know how to determine whether libopencv_imgcodecs.so exists or the version of OpenCV is 2.